sklearn/naive_bayes.py

```diff
@@ -117,6 +117,229 @@ def predict_proba(self, X):
         return np.exp(self.predict_log_proba(X))
 
 
+class GeneralNB(_BaseNB, _BaseComposition, ClassifierMixin):
+    """General Naive Bayes
+
+    Read more in the :ref:`User Guide <general_naive_bayes>`.
+
+    Attributes
+    ----------
+    # TODO
+    distributions : list of tuples
+        List of (name, distribution, column(s)) tuples specifying the
+        distribution objects to be applied to subsets of the data.
+
+        name : string
+            Like in Pipeline and ColumnTransformer, this allows the
+            distribution and its parameters to be set using ``set_params``.
+        distribution : estimator or {'passthrough', 'drop'}
+            Estimator must support :term:`fit` and :term:`transform`.
+            Special-cased strings 'drop' and 'passthrough' are accepted as
+            well, to indicate to drop the columns or to pass them through
+            untransformed, respectively.
+        column(s) : string or int, array-like of string or int, slice, \
+boolean mask array or callable
+            Indexes the data on its second axis. Integers are interpreted as
+            positional columns, while strings can reference DataFrame columns
+            by name.  A scalar string or int should be used where
+            ``transformer`` expects X to be a 1d array-like (vector),
+            otherwise a 2d array will be passed to the transformer.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.naive_bayes import GeneralNB, GaussianNB, BernoulliNB
+    >>> X = np.array([[1.5, 2.3, 5.7, 0, 1],
+    ...               [2.7, 3.8, 2.3, 1, 0],
+    ...               [1.7, 0.1, 4.5, 1, 0]])
+    >>> y = np.array([1, 0, 0])
+    >>> clf = GeneralNB([
+    >>>     ("gaussian", GaussianNB(), [0, 1, 2]),
+    >>>     ("bernoulli", BernoulliNB(), [3, 4])
+    >>> ])
+    >>> clf.fit(X, y)
+    >>> print(clf.predict([[1.5, 2.3, 5.7, 0, 1]]))
+    [1]
+    >>> print(clf.score([[2.7, 3.8, 1, 0, 1]],[0]))
+    1.0
+    """
+    # TODO check_is_fitted
+    # TODO consider jll for each estimator
+
+    def __init__(self, distributions):
+        self.distributions = distributions
+        self._fits = []
+        self._is_fitted = False
+
+    @property
+    def distributions_(self):
+        return [(name, distr) for name, distr, _ in self.distributions]
+
+    @distributions_.setter
```

Comment:
I see why this might be helpful, but we would usually require the user to fit the model to set a `_`-suffixed attribute.

Reply:
Would you say we drop this attribute?

Reply:
It's unconventional in this library to provide a setter. But we should keep the distributions_ attribute like we have the transformers_ attribute in ColumnTransformer

Reply:
Okay, dropping the getter and setter for now.